### PR TITLE
Refactored columnSorting plugin - stage 2

### PR DESCRIPTION
**Pull request by wszymanski**
### Context
- [x] Improved code readability 
- [x] Changed code to meet propositions from previous CR: https://github.com/handsontable/handsontable/pull/5057#discussion_r184086098, https://github.com/handsontable/handsontable/pull/5057#discussion_r184637464), https://github.com/handsontable/handsontable/pull/5057#discussion_r184641939
- [x] Fixed bug #3900 (also related to comment https://github.com/handsontable/handsontable/pull/5057#discussion_r184647806)

### How has this been tested?
Manual tests on column sorting, including changing the order of columns.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #5064 
2. #3900

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/plugins/columnSorting/columnSorting.js [hunk 1]
```diff
@@ -154,30 +152,52 @@ class ColumnSorting extends BasePlugin {
    * @fires Hooks#beforeColumnSort
    * @fires Hooks#afterColumnSort
    */
-  sort(column, order) {
-    this.setSortingColumn(column, order);
+  sort(column, order = this.getNextOrderState(column)) {
+    const allowSorting = this.hot.runHooks('beforeColumnSort', column, order);
```
Comment (swistach) [inline src/plugins/columnSorting/columnSorting.js]: Is it necessary to run a hook, even if a column is `undefined`?

File: src/plugins/columnSorting/columnSorting.js [hunk 2]
```diff
@@ -154,30 +152,52 @@ class ColumnSorting extends BasePlugin {
...
+    if (this.sortColumn === physicalColumn) {
+      if (this.sortOrder === DESC_SORT_STATE) {
+        return NONE_SORT_STATE;
+
+      } else if (this.sortOrder === ASC_SORT_STATE) {
+        return DESC_SORT_STATE;
+      }
+    }
+
+    return ASC_SORT_STATE;
```
Comment (swistach) [inline src/plugins/columnSorting/columnSorting.js]: Could you use `return` once? With the cached result in a variable.
@budnix mentioned that will be useful when we implement automatic code coverage.

File: src/plugins/columnSorting/columnSorting.js [hunk 3]
```diff
@@ -292,8 +273,9 @@ class ColumnSorting extends BasePlugin {
     }
 
     const indexesWithData = [];
-    const columnMeta = this.hot.getCellMeta(0, this.sortColumn);
-    const sortFunction = this.getSortFunctionForColumn(columnMeta);
+    const visualColumn = this.hot.toVisualColumn(this.sortColumn);
+    const columnMeta = this.hot.getCellMeta(0, visualColumn);
+    const sortFunction = getSortFunctionForColumn(columnMeta);
     const emptyRows = this.hot.countEmptyRows();
     let nrOfRows;
```
Comment (swistach) [inline src/plugins/columnSorting/columnSorting.js]: Maybe `nrOfRows` could be `const`?

File: src/plugins/columnSorting/test/columnSorting.e2e.js [hunk 4]
```diff
@@ -61,6 +61,41 @@ describe('ColumnSorting', () => {
...
+        return 1;
+
+      } else if (column === 1) {
+        return 0;
+      }
+
+      return column;
+    };
+
+    const hot = handsontable({
```
Comment (swistach) [inline src/plugins/columnSorting/test/columnSorting.e2e.js]: You can use only `handsontable({})`.

File: src/plugins/columnSorting/test/columnSorting.e2e.js [hunk 5]
```diff
@@ -61,6 +61,41 @@ describe('ColumnSorting', () => {
...
+      ],
+      colHeaders: true,
+      columnSorting: true,
+      sortIndicator: true
+    });
+
+    getPlugin('ColumnSorting').sort(0, 'asc');
+
+    // changing column order: 0 <-> 1
+    hot.updateSettings({modifyCol});
```
Comment (swistach) [inline src/plugins/columnSorting/test/columnSorting.e2e.js]: Just `updateSettings()` should be called in proper context.

File: src/plugins/columnSorting/test/columnSorting.e2e.js [hunk 6]
```diff
@@ -61,6 +61,41 @@ describe('ColumnSorting', () => {
...
+
+    const sortedColumn = this.$container.find('th span.columnSorting')[1];
+    const afterValue = window.getComputedStyle(sortedColumn, ':after').getPropertyValue('content');
+
+    expect(afterValue.indexOf(String.fromCharCode(9650))).toBeGreaterThan(-1);
```
Comment (swistach) [inline src/plugins/columnSorting/test/columnSorting.e2e.js]: Maybe you can cache `String.fromCharCode(9650)` once at a very beginning, with a comment, what char it should be.

File: src/plugins/columnSorting/columnSorting.js (hunk 2 above)
Comment (wszymanski) [reply src/plugins/columnSorting/columnSorting.js]: For discussion with @jansiegel. I will remember that.

**Pull request merged**